bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-items.js

```diff
@@ -11,20 +11,49 @@ import { FieldThingPicker } from './fields/thing-field'
 export default function (f7, isGraalJs) {
   /* Helper block to allow selecting an item */
   Blockly.Blocks['oh_item'] = {
+    fieldPicker: null,
     init: function () {
+      this.fieldPicker = new FieldItemModelPicker('MyItem', null, { f7 })
       this.appendDummyInput()
         .appendField('item')
-        .appendField(new FieldItemModelPicker('MyItem', null, { f7 }), 'itemName')
+        .appendField(this.fieldPicker, 'itemName')
       this.setColour(160)
       this.setInputsInline(true)
-      this.setTooltip('Pick an item from the Model')
+
+      let thisFieldpicker = this.fieldPicker
+      this.setTooltip(function () {
+        let tooltip = 'Pick an item from the Model'
+        const itemData = thisFieldpicker.data
+        if (itemData[0] !== itemData[1]) {
+          tooltip = itemData[0]
+        }
+        return tooltip
+      })
       this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-items-things.html#item')
       this.setOutput(true, 'oh_item')
+    },
+    update_: function (name, label) {
+      this.fieldPicker.data = [name, label]
+    },
+    mutationToDom: function () {
+      let container = Blockly.utils.xml.createElement('mutation')
+
+      if (!this.fieldPicker.data) { // "migrate" old storage
+        this.fieldPicker.data = [this.fieldPicker.value_, this.fieldPicker.value_]
```

Comment:
We could get the label from the REST API:

```suggestion
        api.get(`/rest/items/${this.fieldPicker.value_}?metadata=^$`).then((data) => {
          this.fieldPicker.data = [this.fieldPicker.value_, data.label]
        }).catch()
```

You need this import on top:

```javascript
import api from '@/js/openhab/api'
```

BUT there are currently other issues when trying to make this work, so please address my review in one commit and for this here an additional commit.

Reply:
To be honest, this wasn't even meant to be committed at all because I failed to get it running. So I will uncomment it for the time being for you to jump in.